eng/config/runtimeconfig.template.json

```diff
@@ -1,3 +1,6 @@
 {
-  "rollForwardOnNoCandidateFx": 2
+  "rollForwardOnNoCandidateFx": 2,
+  "configProperties": {
+    "System.GC.Server": true
```

Comment:
@vzarytovskii You mentioned that you thought arcade would handle server mode - it looks like it needs to go in this config file.

Reply:
However I'm sceptical that this is the runtimeconfig.template.json used for the final `fsc.dll` put into the .NET SDK, because this setting `rollForwardOnNoCandidateFx` doesn't seem to appear in the SDK `fsc.runtimeconfig.json`, which currently looks like this:

C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\FSharp\fsc.runtimeconfig.json
```
{
  "runtimeOptions": {
    "tfm": "net7.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "7.0.0-preview.7.22375.6"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
```

Reply:
The `runtimeconfig.template.json` to use is specified here:
```
<Project>
  <PropertyGroup>
    <!--
      When building a .NET Core exe make sure to include the template runtimeconfig.json file
      which has all of our global settings
    -->
    <UserRuntimeConfig Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(OutputType)' == 'Exe'">$(RepositoryEngineeringDir)config\runtimeconfig.template.json</UserRuntimeConfig>
  </PropertyGroup>
</Project>
```
Presumably this config changes when building the actual .NET SDK.  I'm concerned that Arcade is not giving us any way to insert the settings we need for `fsc`

Reply:
all of the 'layout' portion of the SDK with respect to F# happens [in this proejct file](https://github.com/dotnet/sdk/blob/main/src/Layout/tool_fsharp/tool_fsc.csproj).

This is used to publish the F# folder by calling the `Publish` target [here](https://github.dev/dotnet/sdk/blob/3dd9fa18ce109bdec23bf84efa4c3aeca72b947c/src/Layout/redist/targets/GenerateLayout.targets#L183-L186).

So I'd look for a way to make that project take the already-templated outputs from dotnet/fsharp if possible?

Reply:
Yeah somehow `fsc.runtimeconfig.json` and `fsi.runtimeconfif.json` are being generated, I'll leave the details for others to work out.  In short we want Server GC mode enabled for the compiler :)